oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospector.java

```diff
@@ -295,4 +300,94 @@ default List<String> getScopes() {
+			return new SpringOpaqueTokenIntrospectorBuilder(introspectionUri);
+		}
+
+		/**
+		 * Creates a {@code SpringOpaqueTokenIntrospector} with the provided parameters
+		 * @param clientId The client id authorized that should be encode
+		 * @param clientSecret The client secret that should be encode for the authorized
+		 * client
+		 * @return the {@link SpringOpaqueTokenIntrospector}
+		 * @since 6.5
+		 */
+		public SpringOpaqueTokenIntrospector introspectionEncodeClientCredentials(String clientId,
+				String clientSecret) {
+			return this.introspectionEncodeClientCredentials(clientId, clientSecret, StandardCharsets.UTF_8);
+		}
+
+		/**
+		 * Creates a {@code SpringOpaqueTokenIntrospector} with the provided parameters
+		 * @param clientId The client id authorized that should be encode
+		 * @param clientSecret The client secret that should be encode for the authorized
+		 * client
+		 * @param charset the charset to use
+		 * @return the {@link SpringOpaqueTokenIntrospector}
+		 * @since 6.5
+		 */
+		public SpringOpaqueTokenIntrospector introspectionEncodeClientCredentials(String clientId, String clientSecret,
+				Charset charset) {
+			Assert.notNull(clientId, "clientId cannot be null");
+			Assert.notNull(clientSecret, "clientSecret cannot be null");
+			Assert.notNull(charset, "charset cannot be null");
+			String encodeClientId = URLEncoder.encode(clientId, charset);
+			String encodeClientSecret = URLEncoder.encode(clientSecret, charset);
+			RestTemplate restTemplate = new RestTemplate();
+			restTemplate.getInterceptors().add(new BasicAuthenticationInterceptor(encodeClientId, encodeClientSecret));
+			return new SpringOpaqueTokenIntrospector(this.introspectionUri, restTemplate);
+		}
+
+		/**
+		 * Creates a {@code SpringOpaqueTokenIntrospector} with the provided parameters
+		 * @param clientId The client id authorized
+		 * @param clientSecret The client secret for the authorized client
+		 * @return the {@link SpringOpaqueTokenIntrospector}
+		 * @since 6.5
+		 */
+		public SpringOpaqueTokenIntrospector introspectionClientCredentials(String clientId, String clientSecret) {
+			Assert.notNull(clientId, "clientId cannot be null");
+			Assert.notNull(clientSecret, "clientSecret cannot be null");
+			RestTemplate restTemplate = new RestTemplate();
+			restTemplate.getInterceptors().add(new BasicAuthenticationInterceptor(clientId, clientSecret));
+			return new SpringOpaqueTokenIntrospector(this.introspectionUri, restTemplate);
+		}
+
+		/**
+		 * Creates a {@code SpringOpaqueTokenIntrospector} with the provided parameters
+		 * The given {@link RestOperations} should perform its own client authentication
+		 * against the introspection endpoint.
+		 * @param restOperations The client for performing the introspection request
+		 * @return the {@link SpringOpaqueTokenIntrospector}
+		 * @since 6.5
+		 */
+		public SpringOpaqueTokenIntrospector withRestOperations(RestOperations restOperations) {
```

Comment:
Let's leave this out for now. If they have a `RestOperations`, they can simply use the `RestOperations` constructor. Since they at that point need to add their own interceptor, I trust that they know what they are doing regarding client id and secret.